### PR TITLE
Hide smtp pass from instance config output

### PR DESCRIPTION
**Pull request by mraerino**
**- Summary**

We do not want to show a once entered password when getting instance details.

**- Test plan**

- Tests pass

**- Description for the changelog**

Hide SMTP password from instance return

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by vbrown608** [comment]
This might be a naive question (I haven't looked at this project much yet), but won't other handlers still return a non-empty SMTP password, eg `UpdateInstance`?

**Comment by mraerino** [comment]
Those are really good points 😄 
Fixed it.

File: api/instance.go [hunk 1]
```diff
@@ -104,6 +112,7 @@ func (a *API) UpdateInstance(w http.ResponseWriter, r *http.Request) error {
 		if err := i.UpdateConfig(a.db, params.BaseConfig); err != nil {
 			return internalServerError("Database error updating instance").WithInternalError(err)
 		}
+		i.BaseConfig.SMTP.Pass = "" // hide from response
```
Comment (vbrown608) [inline api/instance.go]: `SMTP.Pass` is still output by this handler if `params.BaseConfig` is nil.

Is there anywhere that we *want* to output the SMTP password to JSON? If not we could do `json:"-"` in the tags. 

Or, if outputting it is just a special case, we could wrap it in a new struct with special tags there and do `json:"-"` by default.

Comment (mraerino) [reply api/instance.go]: good catch. Thank you!

I thought about the json tag, but it really seems to complicated

**Pull request merged**